cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/sidebar_filtering.spec.js

```diff
@@ -0,0 +1,303 @@
+  const fourthTestField = testFields[3];
+  cy.getElementByTestId('fieldToggle-' + thirdTestField).click();
+  cy.getElementByTestId('fieldToggle-' + fourthTestField).click();
+  getDocTableHeaderByIndex(1).should('have.text', '_source');
+  getDocTableHeaderByIndex(2).should('not.exist');
+  // Select all test fields
+  selectFields(testFields);
+  // Check default column again
+  getDocTableHeaderByIndex(1).should('not.have.text', '_source');
+  // Check the columns match the selected fields
+  checkTableHeadersByArr(testFields);
+  if (indexPattern) {
+    // Validate default hits
+    cy.getElementByTestId('discoverQueryHits').should('have.text', '10,000');
+  }
+  // Send PPL query
+  cy.intercept('**/api/enhancements/search/ppl').as('pplQuery');
+  dataExplorer.sendQueryOnMultilineEditor(pplQuery);
+  cy.wait('@pplQuery').then(() => {
+    // Check table headers persistence after PPL query
+    cy.wait(1000);
+    checkTableHeadersByArr(testFields);
+    if (indexPattern) {
+      // Check filter was correctly applied
+      cy.getElementByTestId('discoverQueryHits').should('have.text', '1,152');
+    }
+    // Validate the first 5 rows on the _id column
+    checkDocTableColumnByArr(expectedValues, 2);
+  });
+  // Send SQL query
+  dataExplorer.setQueryEditorLanguage('OpenSearch SQL');
+  cy.intercept('**/api/enhancements/search/sql').as('sqlQuery');
+  dataExplorer.sendQueryOnMultilineEditor(sqlQuery);
+  cy.wait('@sqlQuery').then(() => {
+    // Check table headers persistence after SQL query
+    cy.wait(1000);
+    checkTableHeadersByArr(testFields);
+    // Validate the first 5 rows on the _id column
+    checkDocTableColumnByArr(expectedValues, 2);
+  });
+  // Clean all test fields for the next test
+  selectFields(testFields);
+};
+
+const checkFilteredFields = (indexPattern = true) => {
+  const filterFields = () => {
+    const expectedValues = [
+      { search: '_index', assertion: 'equal' },
+      { search: ' ', assertion: null }, // no field should contain spaces
+      { search: 'a', assertion: 'include' },
+      { search: 'ag', assertion: 'include' },
+      { search: 'age', assertion: 'include' },
+      { search: 'non-existent field', assertion: null },
+    ];
+    expectedValues.forEach(({ search, assertion }) => {
+      dataExplorer.checkSidebarFilterBarResults(search, assertion);
+    });
+  };
+
+  if (indexPattern) {
+    dataExplorer.selectIndexPatternDataset(INDEX_PATTERN_NAME, 'DQL');
+    dataExplorer.setQueryEditorLanguage('DQL');
+    cy.setTopNavDate(START_TIME, END_TIME);
+    filterFields();
+    dataExplorer.setQueryEditorLanguage('Lucene');
+    filterFields();
+  } else {
+    dataExplorer.selectIndexDataset(DATASOURCE_NAME, INDEX_NAME, 'PPL', 'timestamp');
+  }
+  dataExplorer.setQueryEditorLanguage('PPL');
+  filterFields();
+  dataExplorer.setQueryEditorLanguage('OpenSearch SQL');
+  filterFields();
+};
+
+const checkCollapseAndExpand = (indexPattern = true) => {
```

Comment:
nitpick: `checkSidebarPanelCollapseAndExpandBehavior = (isIndexPattern)` (i personally wouldn't provide a defaultValue for the argument for this)